devel-common/src/tests_common/test_utils/api_client_helpers.py

```diff
@@ -49,3 +56,24 @@ def generate_access_token(username: str, password: str, host: str) -> str:
 
     assert access_token, f"Failed to get JWT token from redirect url {url} with status code {login_response}"
     return access_token
+
+
+def make_authenticated_rest_api_request(
+    path: str,
+    method: str,
+    body: dict | None = None,
+    api_username: str = "admin",
+    api_password: str = "admin",
+    api_server_url: str | None = None,
+):
+    if not api_server_url:
+        api_server_url = conf.get("api", "base_url", fallback="http://localhost:8080")
+    api_server_url = api_server_url.rstrip("/")
+    token = generate_access_token(api_username, api_password, api_server_url)
```

Comment:
As mentioned in Slack, we might want to set `export AIRFLOW__CORE__SIMPLE_AUTH_MANAGER_ALL_ADMINS="True"` in AWS CI and call the Airflow API with no credentials. That would simplify things

Reply:
I tried this but it still needs a valid token. I think it just makes all users admin but you still have to login and use the token

Reply:
Interesting, you sure you tested with this change: #49599? That's literally the goal of this PR :) To automatically include a token in the request if this config is on

Reply:
Ohh I just tried removing the header completely instead of sending an empty token and it worked!

Reply:
Nice :)